### DISCUSSION TITLE
v9.1: fallback to WebGL where WebGPU is not supported?

### DISCUSSION THREAD

**Discussion by TextTL** [visgl/deck.gl#9171] [General]
Suppose that when deck.gl 9.1  is completed. If someone upgrades to v9.1. The development environment is on VMs (such as vmware), where VMWare sVGA supports only webgl2, not yet supports WebGPU. it possible for deck.gl automatically falling back to webgl2 when navigator.gpu.requestAdapter() fails?

**Comment by ibgreen** [comment]
Yes, WebGL will still be supported and you will be able to configure which GPU backend to use, the default setting will be `best-available`, meaning WebGPU if available and WebGL if not.

FWIW, the official support policy for these frameworks are that running on VMs is not supported. It is just a limit to what we officially support, we don't actively block it and would accept PRs with reasonable fixes. I.e. if it works for you and you are willing to fix issues yourself you may well be OK, but we definitely won't be debugging issues in VMs.

**Reply by TextTL** [reply]
Thank you!
